palette/src/matrix.rs

```diff
@@ -103,23 +125,24 @@ pub fn rgb_to_xyz_matrix<S: RgbSpace, T: FloatComponent>() -> Mat3<T> {
     let g: Xyz<S::WhitePoint, T> = S::Primaries::green().into_color_unclamped();
     let b: Xyz<S::WhitePoint, T> = S::Primaries::blue().into_color_unclamped();
 
-    let mut transform_matrix = mat3_from_primaries(r, g, b);
+    let [t0, t1, t2, t3, t4, t5, t6, t7, t8] = mat3_from_primaries(r, g, b);
 
     let s_matrix: Rgb<Linear<S>, T> = multiply_xyz_to_rgb(
-        &matrix_inverse(&transform_matrix),
+        &matrix_inverse(&[t0, t1, t2, t3, t4, t5, t6, t7, t8]),
         &S::WhitePoint::get_xyz(),
     );
-    transform_matrix[0] = transform_matrix[0] * s_matrix.red;
-    transform_matrix[1] = transform_matrix[1] * s_matrix.green;
-    transform_matrix[2] = transform_matrix[2] * s_matrix.blue;
-    transform_matrix[3] = transform_matrix[3] * s_matrix.red;
-    transform_matrix[4] = transform_matrix[4] * s_matrix.green;
-    transform_matrix[5] = transform_matrix[5] * s_matrix.blue;
-    transform_matrix[6] = transform_matrix[6] * s_matrix.red;
-    transform_matrix[7] = transform_matrix[7] * s_matrix.green;
-    transform_matrix[8] = transform_matrix[8] * s_matrix.blue;
-
-    transform_matrix
+
+    [
+        t0 * s_matrix.red,
+        t1 * s_matrix.green,
+        t2 * s_matrix.blue,
+        t3 * s_matrix.red,
+        t4 * s_matrix.green,
+        t5 * s_matrix.blue,
+        t6 * s_matrix.red,
+        t7 * s_matrix.green,
+        t8 * s_matrix.blue,
+    ]
```

Comment:
I noticed an improvement for this, will have to see if it's just my machine or not.